index.js

```diff
@@ -1,2 +1,118 @@
 #!/usr/bin/env node
-require('./build/deploy')
+'use strict'
+
+const cloneDeep = require('lodash.cloneDeep')
+const co = require('co')
+const fs = require('mz/fs')
+const inquirer = require('inquirer')
+const path = require('path')
+const rsyncwrapper = require('rsyncwrapper')
+const yargs = require('yargs')
+
+function rsync(opts) {
+	return new Promise((yes, no) => {
+		rsyncwrapper(opts, (err, stdout, stderr, cmd) => {
+			const ret = { err, stdout, stderr, cmd }
+			return err ? no(ret) : yes(ret)
+		})
+	})
+}
+
+const interactive = co.wrap(function* (conf) {
+	let destIndex = 0
+	if (conf.destinations.length > 1) {
+		const answers = yield inquirer.prompt([{
+			name: 'destIndex',
+			type: 'list',
+			message: 'Which host do you want to deploy to?',
+			choices: conf.destinations.map((dest, index) => ({ name: dest.name, value: index }))
+		}])
+		destIndex = answers.destIndex
+	}
+
+	const typeArgs = yield inquirer.prompt([ {
+		type: 'list',
+		name: 'type',
+		message: 'How would you like to deploy?',
+		choices: [
+			{ name: 'Normal (--dry-run)', value: { dryRun: true } },
+			{ name: 'Normal', value: [] },
+			{ name: 'Delete (--delete --dry-run)', value: { delete: true, dryRun: true } },
+			{ name: 'Delete (--delete)', value: { delete: true } } ]
+	} ])
+
+	return {
+		destIndex,
+		delete: typeArgs.type.delete || false,
+		dryRun: typeArgs.type.dryRun || false,
+	}
+})
+
+//
+// MAIN
+//
+co(function* () {
+	const pkgPath = path.join(process.cwd(), 'package.json')
+	const args = yargs.usage('$0 [options]')
+		.options({
+			'alias': { alias: 'a', type: 'string', desc: '' },
+			'delete': { alias: 'd', type: 'boolean', default: false },
+			'no-confirm': { alias: 'f', type: 'boolean', default: false },
+			'dry-run': { alias: 'n', type: 'boolean', default: false },
+		})
+		.help().alias('help', 'h')
+		.argv
+
+	yield fs.access(pkgPath, fs.R_OK)
+
+	let conf = cloneDeep(require(pkgPath).rsync)
```

Comment:
Since we're not writing back to `package.json`, I have no problem getting rid of the `cloneDeep` dependency, and just modifying `conf` directly; if that's the case we could do:

Either way I don't think we need `cloneDeep`, because we can switch this to:
```js
const conf = Object.assign({ }, require(pkgPath).rsync, { destinations: [ ], args: [ ], src: './' })
```
The first `{ }` makes sure that all assignments get applied to a new object.